tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -50,8 +53,48 @@ class SourceMapWriter(
   private val _nameToIndex = new HashMap[String, Int]
 
   // Strings are nullable in this stack
-  private val nodePosStack = new Stack[(Position, String)]
-  nodePosStack.push((NoPosition, null))
+  private val nodePosStack = new NodePosStack
+  nodePosStack.push(NoPosition, null)
+
+  private final class NodePosStack {
+    private var topIndex: Int = -1
+    private var posStack: Array[Position] = new Array(32)
+    private var nameStack: Array[String] = new Array(32)
+
+    def pop(): Unit = {
+      val topIdx = topIndex
+      val newTopIdx = topIdx - 1
+      topIndex = newTopIdx
+      posStack(topIdx) = null
+      nameStack(topIdx) = null
```

Comment:
Nulling out stuff is just wasting cycles here. The strings and positions cannot be reclaimed anyway because they're still in the `Tree`. And the stack itself will be freed before the tree is removed.